components/SignIn.tsx

```diff
@@ -83,6 +85,13 @@ export default function SignIn({
           </button>
         </div>
       </div>
+      <SignUpModal
```

Comment:
I noticed you moved the `SignUpModal` away from `components/Layout.tsx` and you're now managing whether its `open` or not through a `useState` in this component, lets not do that. I moved all modals to `components/Layout.tsx` so we could manage their `open` states through the `ModalProvider`.

You should only ever have to use:
```ts
const modals = useModalContext()

modals.open(Modal.SignUp)
modals.close(Modal.SignUp)
```

By moving it here and managing its state with a local `useState` you are breaking the `ModalProvider`.

Reply:
in this case how do I also provide it an optional component to know whether or not this should run `saveAndReturn()` with `onSignUpComplete` using the saveProgress button vs when logging in via the user button in the NavBar

Reply:
The system did not support it yet so I added it in [this PR](https://github.com/saving-satoshi/saving-satoshi/pull/430) but you can pass a second argument to `open`, e.g. 

```ts
modals.open(Modals.SignUp, { a: 1 })
```

Then in `SignUp.tsx` you can do:

```ts
export default function SignUpModal({ state, onClose }) {
  const { a } = state.meta
  console.log(a) // Outputs: 1
}